### PR TITLE
PMM-9375-explaning-api-keys

### PR DESCRIPTION
**Pull request by rasika-chivate**
Update the document with the steps to generate and use API keys...

### CODE REVIEW

File: docs/details/api.md [hunk 1]
```diff
@@ -19,3 +19,45 @@ The objects visible are nodes, services, and agents:
...
+6. *API Key Created* window displays your newly created key. Make sure to copy your key and keep it secure. 
+
+### Authenticate
+
+You can authenticate your request using the HTTP header.
+
+**Example**
+
+```sh
+curl -H "Authorization: Bearer <api_key> http://127.0.0.1/graph/api/dashboards/home
```
Comment (denisok) [inline docs/details/api.md:51]:
```suggestion
curl -H "Authorization: Bearer <api_key>" http://127.0.0.1/graph/api/dashboards/home
```

Comment (denisok) [reply docs/details/api.md:51]: it should work, but in our case it doesn't:
```
curl: (3) URL using bad/illegal format or missing URL
curl: (60) SSL certificate problem: self signed certificate
More details here: https://curl.se/docs/sslcerts.html

curl failed to verify the legitimacy of the server and therefore could not
establish a secure connection to it. To learn more about this situation and
how to fix it, please visit the web page mentioned above.
```
so we need to pass `-k` additionally.

Comment (rnovikovP) [inline docs/details/api.md:51]:
```suggestion
curl -H "Authorization: Bearer <api_key> http://127.0.0.1/v1/version
```

Maybe some  PMM API example is better. I don't expect people much use  Grafana APIs

**Review by rnovikovP — APPROVED** [review]
LGTM, but see a comment about the example we'll use

File: docs/details/api.md [hunk 2]
```diff
@@ -19,3 +19,45 @@ The objects visible are nodes, services, and agents:
...
+
+### Use an API key
+
+You can pass the API key into a REST API call as a query parameter in the following format. Replace `api_key` with your API key.
+
+**Example**
+
+
+```sh
+curl -X GET https://api_key:API_KEY@localhost/v1/version
```
Comment (denisok) [inline docs/details/api.md:62]: same here, we need `-k` option:
```
[dkondratenko@denlen tmp]$ curl -X GET https://api_key:eyJrIjoiYVcyQWlSZ3dMaHhIS09ydHRzTlV0SVB2dERwckozNkIiLCJuIjoicG1tIiwiaWQiOjF9@192.168.49.2:30443/v1/version
curl: (60) SSL certificate problem: self signed certificate
More details here: https://curl.se/docs/sslcerts.html

curl failed to verify the legitimacy of the server and therefore could not
establish a secure connection to it. To learn more about this situation and
how to fix it, please visit the web page mentioned above.
[dkondratenko@denlen tmp]$ curl -k -X GET https://api_key:eyJrIjoiYVcyQWlSZ3dMaHhIS09ydHRzTlV0SVB2dERwckozNkIiLCJuIjoicG1tIiwiaWQiOjF9@192.168.49.2:30443/v1/version
{
  "version": "2.25.0",
  "server": {
    "version": "2.25.0",
    "full_version": "2.25.0-64.2112130941.7f24595.el7",
    "timestamp": "2021-12-13T09:41:40Z"
  },
  "managed": {
    "version": "2.25.0",
    "full_version": "dc091abb30f1e0ac8769a02d0587aa1be653a58c",
    "timestamp": "2021-12-13T09:39:25Z"
  },
  "distribution_method": "DOCKER"```

Comment (denisok) [reply docs/details/api.md:62]: @rnovikovP what should we do - ask ppl to use insecure option or provide good instructions but fix cert issues?

**Review by denisok — CHANGES_REQUESTED** [review]

File: docs/details/api.md (hunk 2 above)
Comment (denisok) [reply docs/details/api.md:62]: or should we have a section about how to install our certificates ? and mention it here?

Comment (rnovikovP) [reply docs/details/api.md:62]: httpS! You right.
And yes, some notes about how to ignore certificates. 

We do the same on `pmm-admin config ...` and providing instructions for Certificates is out of the scope of this PR

File: docs/details/api.md [hunk 3]
```diff
@@ -19,3 +19,45 @@ The objects visible are nodes, services, and agents:
...
+
+**Example**
+
+```sh
+curl -H "Authorization: Bearer <api_key> http://127.0.0.1/graph/api/dashboards/home
+```
+
+### Use an API key
+
+You can pass the API key into a REST API call as a query parameter in the following format. Replace `api_key` with your API key.
```
Comment (BupycHuk) [inline docs/details/api.md:56]:
```suggestion
You can pass the API key into a REST API call as a query parameter in the following format. Replace `API_KEY` with your API key.
```

File: docs/details/api.md [hunk 4]
```diff
@@ -19,3 +19,45 @@ The objects visible are nodes, services, and agents:
...
+
+You can authenticate your request using the HTTP header.
+
+**Example**
+
+```sh
+curl -H "Authorization: Bearer <api_key> http://127.0.0.1/graph/api/dashboards/home
+```
+
+### Use an API key
```
Comment (BupycHuk) [inline docs/details/api.md:54]:
```suggestion
### Use an API key in basic auth
```

File: docs/details/api.md (hunk 1 above)
Comment (denisok) [reply docs/details/api.md:51]: my bad in example it is http - it works OK.

**Review by puneet0191 — APPROVED** [review]

File: docs/details/api.md [hunk 5]
```diff
@@ -46,7 +46,7 @@ PMM uses the Grafana API keys for authentication. Following are the steps to gen
 You can authenticate your request using the HTTPS header.
 
 !!! caution alert alert-warning "Important"
-    Use the `-k` or `--insecure` option to force cURL to ignore invalid and self-signed SSL certificate errors. The option will skip the SSL verification process, and you can bypass any SSL errors while still having SSL-encrypted communication.
+    Use the `-k` or `--insecure` parameter to force cURL to ignore invalid and self-signed SSL certificate errors. The option will skip the SSL verification process, and you can bypass any SSL errors while still having SSL-encrypted communication. However, using the `--insecure`  parameter is not recommended. Although the data transfer is encrypted, it is not entirely secure. For enhanced security of your PMM installation, you need valid SSL certificates. For information on validating SSL certificates, visit: https://github.com/percona/pmm-doc/blob/decdf601c580174c7daa245e12b3c06e4f1e6319/docs/how-to/secure.md
```
Comment (denisok) [inline docs/details/api.md:49]: that link wouldn't work, otherwise good

Comment (rasika-chivate) [reply docs/details/api.md:49]: Yes, I will change to relative path.

**Review by denisok — APPROVED** [review]

**Review by BupycHuk — APPROVED** [review]

**Review by rnovikovP — APPROVED** [review]

**Pull request merged**